### ISSUE TITLE
Pyupgrade failing on latest commit 

### ISSUE THREAD

**Issue by alvations** [nltk/nltk#3273]
Something seems to be off on our pre-commit hook with `pyupgrade` (@asottile). 

Did a force push of latest commit and our CI/CD is failing. https://github.com/nltk/nltk/actions/runs/9822351687/job/27119009274 

Also affecting any new PR, e.g. https://github.com/nltk/nltk/pull/3270/files (we might have to remove it to temporarily unblock us).

**Comment by asottile** [comment]
add --show-diff-on-failure and maybe it'll show what's up?

**Comment by alvations** [comment]
Thanks @asottile for the quick reply!

Is that a pyupgrade argument? Getting this locally 

```
pyupgrade: error: unrecognized arguments: --show-diff-on-failure
```

and also in precommit runs https://github.com/nltk/nltk/actions/runs/9822501001/job/27119306847

**Comment by alvations** [comment]
This run showed something, https://github.com/nltk/nltk/actions/runs/9822519340/job/27119340852?pr=3274 

```
pyupgrade................................................................Failed
- hook id: pyupgrade
- exit code: 1
- files were modified by this hook

Rewriting nltk/metrics/aline.py
Rewriting nltk/test/unit/test_corpora.py
```

But somehow my local environment isn't rewriting, I'll check further.

**Comment by alvations** [comment]
@asottile I think I found the problem, you might want to look at `pyupgrade` using miniconda environment, somehow pyupgrade don't rewrite files in miniconda environments with `conda 4.12.0`.  Hope that info helps! 

----

Also posted the related bug on the asottile/pyupgrade#954, gonna disable it on our repo until we can get the issue 954 sorted out.

**Comment by alvations** [comment]
Thank you again @purificant @asottile for the help!

Resolved in #3280
